packages/foam-core/test/markdown-provider.test.ts

```diff
@@ -49,6 +53,24 @@ describe('Markdown loader', () => {
   });
 });
 
+describe('Note Title', () => {
+  it('should initialize note title if heading exists', () => {
+    const graph = new NoteGraph();
+    graph.setNote(createNoteFromMarkdown('page-a', pageA));
+
+    const pageANoteTitle = (graph.getNote('page-a') as Note).title;
+    expect(pageANoteTitle).toBe('Page A');
+  });
+
+  it('should not initialize note title if heading does not exist', () => {
+    const graph = new NoteGraph();
+    graph.setNote(createNoteFromMarkdown('page-d', pageD));
```

Comment:
do you need to add the note to the graph to run this test? (not adding it might also spare you from the casting below)

or is there something else that you are trying to test as well?

Reply:
I guess it's not strictly necessary, but I don't mind doing a little integration testing here to exercise the setNote/getNote to make sure they work on notes without titles, so I'm fine with this.